filestore.go

```diff
@@ -0,0 +1,282 @@
+// An empty name returns a general schema suitable for any file.
+func CreateFileSchemaFor(tdbCtx *Context, name string) (*ArraySchema, error) {
+	var fileURI *C.char
+	if name != "" {
+		fileURI = C.CString(name)
+		defer C.free(unsafe.Pointer(fileURI))
+	}
+
+	arraySchema := ArraySchema{context: tdbCtx}
+	ret := C.tiledb_filestore_schema_create(tdbCtx.tiledbContext, fileURI, &arraySchema.tiledbArraySchema)
+	if ret != C.TILEDB_OK {
+		return nil, fmt.Errorf("Error creating schema: %s", tdbCtx.LastError())
+	}
+	runtime.SetFinalizer(&arraySchema, func(arraySchema *ArraySchema) {
+		arraySchema.Free()
+	})
+
+	return &arraySchema, nil
+}
+
+// CreateGeneralFileSchema returns a general schema for the storage of a file.
+func CreateGeneralFileSchema(tdbCtx *Context) (*ArraySchema, error) {
+	return CreateFileSchemaFor(tdbCtx, "")
+}
+
+// File represents a TileDB filestore file.
+type File struct {
+	tdbCtx   *Context // the tiledb context for all operations
+	arrayURI string   // the uri of the array
+}
+
+// NewFile returns a file. The arrayURI is the uri of the underlying TileDB array that contains the file (see Export) or
+// in which the file will be stored (see Import)
+func NewFile(tdbCtx *Context, arrayURI string) *File {
+	return &File{
+		tdbCtx:   tdbCtx,
+		arrayURI: arrayURI,
+	}
+}
+
+// Size returns the uncompressed size of the file.
+func (f *File) Size() (int64, error) {
+	uri := C.CString(f.arrayURI)
+	defer C.free(unsafe.Pointer(uri))
+
+	var size C.size_t
+	ret := C.tiledb_filestore_size(f.tdbCtx.tiledbContext, uri, &size)
+	if ret != C.TILEDB_OK {
+		return 0, fmt.Errorf("Error getting file size: %s", f.tdbCtx.LastError())
+	}
+
+	return int64(size), nil
+}
+
+// Export writes the contents of f to the named local file.
+func (f *File) Export(name string) error {
+	arrayURI := C.CString(f.arrayURI)
+	defer C.free(unsafe.Pointer(arrayURI))
+	fileURI := C.CString(name)
+	defer C.free(unsafe.Pointer(fileURI))
+
+	ret := C.tiledb_filestore_uri_export(f.tdbCtx.tiledbContext, fileURI, arrayURI)
+	if ret != C.TILEDB_OK {
+		return fmt.Errorf("Error exporting file: %s", f.tdbCtx.LastError())
+	}
+
+	return nil
+}
+
+// Import replaces the contents of f with the contents of the named local file.
+// Assumes that the TileDB array exists and is of the appropriate schema.
+func (f *File) Import(name string, mimeType FileStoreMimeType) error {
+	arrayURI := C.CString(f.arrayURI)
+	defer C.free(unsafe.Pointer(arrayURI))
+	fileURI := C.CString(name)
+	defer C.free(unsafe.Pointer(fileURI))
+
+	ret := C.tiledb_filestore_uri_import(f.tdbCtx.tiledbContext, arrayURI, fileURI, C.tiledb_mime_type_t(mimeType))
+	if ret != C.TILEDB_OK {
+		return fmt.Errorf("Error importing file: %s", f.tdbCtx.LastError())
+	}
+
+	return nil
+}
+
+// CreateAndImport creates the array, initializes it with the appropriate schema and imports the file.
+func (f *File) CreateAndImport(name string, mimeType FileStoreMimeType) error {
```

Comment:
Would you mind explaining the thinking behind placing all of these functions of the file struct? 

My thinking is the same as above, I think this should be an exported single func that returns a file that is imported and ready for use. The file struct as it is currently written with a constructor doesn't really "mean anything" since you can't really do anything with the file until import/export/create and import are executed.